src/main/java/com/google/growpod/controllers/GardenDao.java

```diff
@@ -74,8 +51,10 @@ public GardenDao(Datastore datastore) {
    * @return the garden with id's data or null.
    */
   public Garden getGardenById(String id) {
-    // MOCK IMPLEMENTATION
-    return GARDEN_MAP.get(id);
+    String projectId = datastoreInstance.getProjectId();
+    Key key = Key.newBuilder(projectId, "Garden", Long.parseLong(id)).build();
+    Entity gardenEntity = datastore.get(key);
+    return gardenEntity == null ? null : Garden.from(gardenEntity);
```

Comment:
Can `Garden.from` return `null`? Is this ternary needed?

Reply:
`datastore.get()` can return `null` when the key does not exist in the database, and `Garden.from()` expects its argument to be non-`null`, so I use the ternary to ensure this doesn't cause any problems.